### ISSUE TITLE
AAiD crashes since newest builds

### ISSUE THREAD

**Issue by fabssor** [hufman/AndroidAutoIdrive#269]
So since the newest builds AAiD crashes after like 10 minutes for me.
Connection works fine but after some time the apps disappear in iDrive.

Setup:
- MyBMW 1.3.4 (6706)f
- AAiD 1.2.2-436-82f316e 

I think this is a problem of AAiD because the apps checkbox in iDrive still is enabled. Also MyBMW still has his connection notification.
I can then reopen AAiD but it crashes after some seconds again. AAiD says in the connection page that the tunnel is up and running.
See the screenshots for more detail.
IDrive Apps is checked:
![WhatsApp Image 2021-05-04 at 20 08 45](https://user-images.githubusercontent.com/30718590/117050277-76900300-ad15-11eb-99ba-5f5210c2bc2d.jpeg)
After AAiD crashed and reopening it, it reports Tunnel is working:
![WhatsApp Image 2021-05-04 at 20 09 17_1](https://user-images.githubusercontent.com/30718590/117050280-77289980-ad15-11eb-9d74-8eb4f576f249.jpeg)
MyBMW notification is showing:
![WhatsApp Image 2021-05-04 at 20 09 17](https://user-images.githubusercontent.com/30718590/117050284-77289980-ad15-11eb-98a9-fd6bb02dfb0b.jpeg)

**Comment by hufman** [comment]
That's a new one I haven't seen before! Thank you for the confirmation screenshots showing that the Apps connection is still connected to the car! I just tried to test it and was listening to music in the car for 30 minutes before My Mini dropped the tunnel, and unfortunately haven't experienced the problem you saw.
I do see a very rare crash reported when inflating some Remote Views from notifications, and so I added error handling around that, but it's rare and I'm not even sure which notification is causing the problem. It happened to 3 people a total of 391 times over 4 months (with one unlucky user having 75% of those occurrences).
I do see some "RHMI Application Already Logged In" crashes, which I'm disappointed about because I thought I fixed all those.
Can you look in `adb logcat` to see what crash messages you are seeing?

**Comment by fabssor** [comment]
Hi hufman, I will provide the logs later today! Thank you :)

**Comment by fabssor** [comment]
So here is the log! Hope it helps out. I would gladly help to find the error but I am no android dev, so I can only support here
 :)
If you need more logs, I may have time tomorrow to pull fresh logs while the app crashes.
[aaid.log](https://github.com/hufman/AndroidAutoIdrive/files/6429509/aaid.log)

**Comment by hufman** [comment]
Yeah the logs look like some more instances of #214  :( I'll try to think of a better solution to it.

**Comment by dimakaob** [comment]
I am sorry if it's the wrong thread. 
My problem is that sometimes (quite often) it just stops playing music. And there is no Spotify in the list, there is no check mark in the "Apps". It helps if i disconnect the phone and connect it again. 
And it's funny, but i noticed, that it stops when i take the phone. Literally take it in my hands.. 
Well, and when i start the car it's also doesn't work often. I suspect it is My BMW's fault. It worked fine a few updates earlier.

**Comment by dimakaob** [comment]
Got it! Thank you for your answer!

**Comment by hufman** [comment]
I just pushed a probable fix, try it out!

**Comment by dimakaob** [comment]
Where can I download it? 😅

пт, 14 мая 2021 г., 23:13 Walter Huf ***@***.***>:

**Comment by hufman** [comment]
I pushed this fix right to the master branch, which you can download from [the link from the main page](https://androidautoidrive.s3.amazonaws.com/hufman/AndroidAutoIdrive/androidautoidrive-latest-master-nomap-sentry-release.apk)

**Comment by dimakaob** [comment]
Thank you so much!!!

пт, 14 мая 2021 г., 23:40 Walter Huf ***@***.***>:

**Comment by dimakaob** [comment]
I don't know what have you done, but so far it works perfectly! Hooray!

**Comment by fabssor** [comment]
Die also not happen for me in a long time. So I think for now we can close this.

**Comment by unlive74** [comment]
The app crashes in 80% of cases if I open the "Car info" tab.
